### DISCUSSION TITLE
yolov8_node 실행 시 에러가 발생합니다.

### DISCUSSION THREAD

**Discussion by ycj0313** [SKKUAutoLab/H-Mobility-Autonomous-Advanced-Course#64] [2024년도 교육 자료 관련 질문] [answered]
![스크린샷 2024-07-15 11-21-11](https://github.com/user-attachments/assets/9c0059f7-2f2d-419e-9964-e921a237c7af)
yolov8_node 실행 시 위와 같은 에러가 발생합니다. 도움 부탁드리겠습니다.

**Comment by ycj0313** [comment]
![스크린샷 2024-07-15 11-23-08](https://github.com/user-attachments/assets/f8884803-3e08-4e12-a186-c4172a33221f)

**Comment by Jinsun-Lee** [comment]
일정이 있어 확인이 늦었습니다. 복귀 후 도움 드리겠습니다. (오후 10시 이전)

**Comment by Jinsun-Lee** [comment, marked as answer]
### 현재 노트북에 설치된 PyTorch와 torchvision의 버전이 다른 것으로 보입니다.
오류 메시지:`RuntimeError: operator torchvision::nms does not exist`

</br>

#### 1. 버전이 호환되는지 확인
![image](https://github.com/user-attachments/assets/8cb91779-e25b-4793-9c0c-9be19c6cbc74)
아래 명령어를 실행하면, 위와 같은 결과를 확인할 수 있습니다.
```
curl -L -O https://github.com/Jinsun-Lee/H-Mobility-Autonomous-Advanced-Course/raw/main/src/debug/check_torch_version.py; python3 check_torch_version.py
```
가장 마지막의 출력 결과가 "PyTorch and torchvision versions are compatible."가 나와야 합니다.  
그 외의 출력이 나오는 경우는 버전을 일치시켜야 합니다.  

</br>



#### 2. 호환 버전 설치
확인한 버전을 참고하여 호환되는 버전을 새로 설치 바랍니다.
새로 설치 후, `python3 check_torch_version.py` 명령으로 버전이 호환되는지 확인 바랍니다.

</br>

```
# 설치 삭제
pip uninstall torch torchvision

# GPU가 있는 경우 실행
pip install torch torchvision
pip install torch torchvision --index-url https://download.pytorch.org/whl/cu121

# PyTorch와 torchvision의 CPU 버전 설치하는 경우 실행
pip install torch torchvision --index-url https://download.pytorch.org/whl/cpu
```

</br>


#### 3. Non-Maximum Suppression을 수행 확인
아래 명령어를 실행하면 PyTorch와 TorchVision을 사용하여 객체 검출 후 Non-Maximum Suppression을 수행하여 인덱스를 추출합니다.  
`tensor([0])`가 출력되는지 확인 바랍니다.   
  
```
curl -L -O https://github.com/Jinsun-Lee/H-Mobility-Autonomous-Advanced-Course/raw/main/src/debug/check_torch_nms.py; python3 check_torch_nms.py
```

**Reply by ycj0313** [reply]
해결하였습니다. 감사합니다!

**Comment by Jinsun-Lee** [comment]
위의 답변에서 `2.3.1+cpu`가 나온 것으로 보아, `pip install torchvision==0.18.1` 명령어를 실행해 보시거나
다운그레이드하여 세팅 바랍니다.
